### ISSUE TITLE
SLA not calculated

### ISSUE THREAD

**Issue by crnfpp** [zammad/zammad#4516]
### Used Zammad Version

5.3

### Environment

- Installation method: Package
- Operating system: Debian GNU/Linux 10 (buster)
- Database + version:  postgresql
- Elasticsearch version: 7.17.9

### Actual behaviour

first_response_in_min
first_response_diff_in_min
close_in_min
close_diff_in_min

are null (nil) instead of being calculated when first response or close happen.

They are only calculated when a ticket is manually reopened and reclosed.

### Expected behaviour

SLA fields to be calculated when first response and closed happen.

### Steps to reproduce the behaviour

1. Create a SLA relative to a specific group.
2. Set escalation times
3. Create a ticket (via mail or api)
4. Check via api (api/v1/tickets/TICKET_ID) that the correct sla id is associated 
5. Create a first article to a ticket
6. Check via api that if _first_response_in_min and first_response_diff_in_min_ are calculated.
7. Close the ticket 
8. Check via api that if _close_in_min and close_diff_in_min_ are calculated. 
9. Check back in a few hours to see if it calculated.

### I'm sure this is a bug and no feature request or a general question.

yes

**Comment by MrGeneration** [comment]
Mind sharing the payloads you're using to achieve this?

**Comment by crnfpp** [comment]
Sure.
Here below you'll fine the lines of our javascript creating the ticket payload and the SLA applied to the ticket.
**The bug affects to our mail 2 ticket flow** 

```

  var article = {
    "subject": subject,
    "reply_to": email,
    "from": firstLastName + " <" + email + ">",
    "body": ticketBody,
    "type": "web",
    "sender_id": 2,
    "internal": false,
    "content_type": "text/html"
  };

  var ticket = {
    "title": subject,
    "group": group,
    "article": article,
    "customer_id": "guess:" + email,
    "tags": tags
  };

  return ticket;
}
function createTicket(ticket, post_url, token, event) {
  var options = {
    "method": "post",
    "contentType": "application/json",
    "headers": { "Authorization": "Token token=" + token },
    "payload": JSON.stringify(ticket),
    "muteHttpExceptions": false
  };


  var httpResponse = UrlFetchApp.fetch(post_url, options);
```


SLAs
```
{
    "calendar_id": 1,
    "name": "Support",
    "first_response_time": 2940,
    "update_time": null,
    "solution_time": 5820,
    "response_time": null,
    "condition": {
      "ticket.group_id": {
        "operator": "is",
        "value": [
          "2",
          "6",
          "4",
          "8",
          "22"
        ]
      }
    },
    "updated_by_id": 4,
    "id": 5,
    "created_by_id": 106,
    "created_at": "2022-02-07T14:46:47.649Z",
    "updated_at": "2023-03-03T10:47:18.559Z",
    "data": {
      
    }
  },
```

**Comment by crnfpp** [comment]
Faulty ticket example
```
{
  "id": 90434,
  "group_id": 25,
  "priority_id": 2,
  "state_id": 4,
  "organization_id": null,
  "number": "20221115210092",
  "title": "_______",
  "owner_id": 8283,
  "customer_id": 31395,
  "note": null,
  "first_response_at": "2022-11-15T16:07:35.886Z",
  "first_response_escalation_at": null,
  "first_response_in_min": null,
  "first_response_diff_in_min": null,
  "close_at": "2022-11-15T16:07:35.747Z",
  "close_escalation_at": null,
  "close_in_min": null,
  "close_diff_in_min": null,
  "update_escalation_at": null,
  "update_in_min": null,
  "update_diff_in_min": null,
  "last_contact_at": "2022-11-15T16:07:35.886Z",
  "last_contact_agent_at": "2022-11-15T16:07:35.886Z",
  "last_contact_customer_at": "2022-11-15T09:50:52.738Z",
  "last_owner_update_at": "2022-11-15T16:07:35.982Z",
  "create_article_type_id": 1,
  "create_article_sender_id": 2,
  "article_count": 2,
  "escalation_at": null,
  "pending_time": null,
  "type": null,
  "time_unit": null,
  "preferences": {
    "channel_id": 26,
    "escalation_calculation": {
      "last_update_at": "2022-11-15T09:50:52.738Z",
      "last_contact_at": "2022-11-15T09:50:52.738Z",
      "sla_id": 5,
      "sla_updated_at": "2022-04-12T08:47:11.281Z",
      "calendar_id": 1,
      "calendar_updated_at": "2022-11-14T16:50:59.573Z",
      "escalation_disabled": false
    }
  },
  "updated_by_id": 8283,
  "created_by_id": 31395,
  "created_at": "2022-11-15T09:50:52.705Z",
  "updated_at": "2022-11-15T16:07:36.026Z",
  "sign": null,
  "last_close_at": null
}
```

**Comment by martini** [comment]
@crnfpp thanks for this additional information. Unfortunately still many infos are missing (or maybe wrong because of your environment - example in Zammad default state_id 4 is closed - which seems to be different on your environment). 

To find out if it's related to your local setup or a general Zammad issue, can you try to verify your behavior on a fresh Zammad installation like one from https://zammad.com/en/getting-started 

Thanks!

**Comment by crnfpp** [comment]
Hi,
thanks, i'll check.
In the mean time here's my ticket states.
It seems like State id 4 is Closed on my env too.

```

// api/v1/ticket_states

[
  {
    "id": 1,
    "state_type_id": 1,
    "name": "new",
    "next_state_id": null,
    "ignore_escalation": false,
    "default_create": true,
    "default_follow_up": false,
    "note": null,
    "active": true,
    "updated_by_id": 1,
    "created_by_id": 1,
    "created_at": "2021-06-01T08:36:22.124Z",
    "updated_at": "2021-06-01T08:36:22.140Z"
  },
  {
    "id": 2,
    "state_type_id": 2,
    "name": "open",
    "next_state_id": null,
    "ignore_escalation": false,
    "default_create": false,
    "default_follow_up": true,
    "note": null,
    "active": true,
    "updated_by_id": 1,
    "created_by_id": 1,
    "created_at": "2021-06-01T08:36:22.137Z",
    "updated_at": "2021-06-01T08:36:22.137Z"
  },
  {
    "id": 3,
    "state_type_id": 3,
    "name": "pending reminder",
    "next_state_id": null,
    "ignore_escalation": true,
    "default_create": false,
    "default_follow_up": false,
    "note": null,
    "active": true,
    "updated_by_id": 1,
    "created_by_id": 1,
    "created_at": "2021-06-01T08:36:22.147Z",
    "updated_at": "2021-06-01T08:36:22.147Z"
  },
  {
    "id": 4,
    "state_type_id": 5,
    "name": "closed",
    "next_state_id": null,
    "ignore_escalation": true,
    "default_create": false,
    "default_follow_up": false,
    "note": null,
    "active": true,
    "updated_by_id": 1,
    "created_by_id": 1,
    "created_at": "2021-06-01T08:36:22.153Z",
    "updated_at": "2021-06-01T08:36:22.153Z"
  },
  {
    "id": 5,
    "state_type_id": 6,
    "name": "merged",
    "next_state_id": null,
    "ignore_escalation": true,
    "default_create": false,
    "default_follow_up": false,
    "note": null,
    "active": true,
    "updated_by_id": 1,
    "created_by_id": 1,
    "created_at": "2021-06-01T08:36:22.158Z",
    "updated_at": "2021-06-01T08:36:22.158Z"
  },
  {
    "id": 6,
    "state_type_id": 7,
    "name": "removed",
    "next_state_id": null,
    "ignore_escalation": true,
    "default_create": false,
    "default_follow_up": false,
    "note": null,
    "active": false,
    "updated_by_id": 1,
    "created_by_id": 1,
    "created_at": "2021-06-01T08:36:22.164Z",
    "updated_at": "2021-06-01T08:36:22.164Z"
  },
  {
    "id": 7,
    "state_type_id": 4,
    "name": "pending close",
    "next_state_id": 4,
    "ignore_escalation": true,
    "default_create": false,
    "default_follow_up": false,
    "note": null,
    "active": true,
    "updated_by_id": 1,
    "created_by_id": 1,
    "created_at": "2021-06-01T08:36:22.170Z",
    "updated_at": "2021-06-01T08:36:22.170Z"
  }
]
```

**Comment by martini** [comment]
> In the mean time here's my ticket states.
> It seems like State id 4 is Closed on my env too.

JFI: Yes, this looks like an regular Zammad installation. Looking forward to have a test environment to reproduce your problem.

**Comment by crnfpp** [comment]
I've create on a trial installation on zammad.com the same SLA (group filter, same time), i've create a token and created a ticket via POST and unfortunately i cannot reproduce the bug.
I'm kind of clueless.

**Comment by crnfpp** [comment]
I was checking the SLAs documentation and i was asking my self if for the close_in_min, close_diff_in_min i need to explicitly specify to consider the state.id 4 in the SLA conditions or the group filter as above is enough.

(I'm looking into SLAs conditions as part of the problem and/or a performance issue at the time of the calculation).

**Comment by MrGeneration** [comment]
Look at the group_ids of your "faulty ticket" and the actual group_ids configured in your SLA.
Zammad is behaving correctly.

----

Sorry, but this seems to be a technical question.
Please note that this repository is the wrong place for technical questions.
Instead, please refer the Zammad community over at: https://community.zammad.org .

In case this turns out the be a bug and no technical question, we'll transfer the relevant information to this repo!

If you require commercial grade support and are no hosted or support contract customer, you can check out our support contracts here: https://zammad.com/pricing#selfhosted
*If you're paying customer already, please consult your Zammad support for assistance! :-)*

**Comment by crnfpp** [comment]
Hi,
I've done few more tests on a fresh installation on a trial installation zammad.com and i've been able to re-produce what could be a bug but i'm not sure.

In bold i've market those steps that maybe a bug or they're code logic.

**Step to reproduce** 

1. I've created two groups (t01 and t02). Users was already present as default group.
2. I've created a SLA01 and applied it to t01
3. Tickets on t01 (via get) now has in preferences sla_id: sla01
4. I've removed from sla 01 the condition "group is t01"
**5. ISSUE: Tickets on t01 still have sla_id: sla01 in preferences.** 
6. I've created SLA02 with conditions "group is t01"
**7. sla_id under preferences are changed in some tickets but not on all tickets** - _is seems like open tickets are updated and not closed? Bug or code logic?_
**8. If i delete SLA02 all tickets that had "preferences>slad_id:2" are not updated and still have "sla_id:2" instead of null.** _Bug or logic?_

So the question is:

- Is it expected that changes of a SLA or Deletion doesn't reflect on ticket?
